spec/controllers/projects_controller_spec.rb

```diff
@@ -70,6 +70,13 @@
     end
   end
 
+  describe 'POST preview' do
+    it 'renders partial preview' do
+      post :preview, project: project.attributes
```

Comment:
Could you rewrite this as

```ruby
xhr :post, :preview, project: project.attributes
```

The `xhr` directive mimics an Ajax request in the controller test, ie. it sets the request format to something javascripty.